### PR TITLE
Updating events

### LINKED ISSUE #13: As a user I want to update a calendar event

### PR DESCRIPTION
**Pull request by Somfic**
#### Description of Change

Adds the ability for an user to update an event in their calendars.

Closes #13.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `cargo test` passes
- [x] tests are changed or added
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] PR release notes describe the change in a way relevant to app developers, and are capitalized, punctuated, and past tense.

#### Release Notes

Notes: 
Ability to update events.

### CODE REVIEW

**Comment by Somfic** [comment]
@KaanMol this is ready for review :)

**Comment by KaanMol** [comment]
At the moment the test names are completely inconsistent throughout the testingframework. Should we fix that first?

**Comment by KaanMol** [comment]
I applied all the names to the format we've discussed outside of GitHub. Can you please go through all the names and check if it turned out like you expected? After that, I will approve the PR and it'll get merged.

File: src/tests/integration/events.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+use crate::{app::events::dto::UpdateEventDto, entity, handlers, tests};
+use actix_web::{test, web};
+
+#[actix_web::test]
+async fn when_update_not_existing_event_expect_error() {
+    let auth_id = "when_update_not_existing_event_expect_error".to_string();
```
Comment (Somfic) [inline src/tests/integration/events.rs:6]: Should this be `when_update_not_existing_event_expect_not_found`?

File: src/tests/integration/sources.rs [hunk 2]
```diff
@@ -43,8 +43,8 @@ async fn test_add_source() {
 }
 
 #[actix_web::test]
-async fn test_add_source_with_invalid_url() {
-    let auth_id = "test_add_source_with_invalid_url".to_string();
+async fn when_add_source_with_invalid_url_expect_error() {
```
Comment (Somfic) [inline src/tests/integration/sources.rs:46]: Should be be expecting a BadFormat error here? `when_add_source_with_invalid_url_expect_bad_format`?

File: src/tests/integration/sources.rs [hunk 3]
```diff
@@ -76,8 +76,8 @@ async fn test_add_source_with_invalid_url() {
 }
 
 #[actix_web::test]
-async fn test_add_source_with_invalid_name() {
-    let auth_id = "test_add_source_with_invalid_url".to_string();
+async fn when_add_source_with_invalid_name_expect_error() {
```
Comment (Somfic) [inline src/tests/integration/sources.rs:79]: Should be be expecting a BadFormat error here? `when_add_source_with_invalid_url_expect_bad_format`?

File: src/tests/unit/events.rs [hunk 4]
```diff
@@ -61,7 +61,7 @@ pub async fn test_get_event_ok() {
 }
 
 #[actix_rt::test]
-pub async fn test_get_event_empty() {
+pub async fn when_get_not_existing_event_expect_error() {
```
Comment (Somfic) [inline src/tests/unit/events.rs:64]: Maybe something along the lines of `when_get_not_existing_event_expect_not_found`?

File: src/tests/unit/sources.rs [hunk 5]
```diff
@@ -34,9 +30,9 @@ pub async fn add_source() {
 }
 
 #[actix_rt::test]
-pub async fn add_source_with_invalid_name() {
+pub async fn when_add_source_with_invalid_name_expect_error() {
```
Comment (Somfic) [inline src/tests/unit/sources.rs:33]: Should be expecting a `ResourecError::InvalidInput` error here? Something like `when_add_source_with_invalid_name_expect_invalid_input` ?

File: src/tests/unit/sources.rs [hunk 6]
```diff
@@ -61,9 +57,9 @@ pub async fn add_source_with_invalid_name() {
 }
 
 #[actix_rt::test]
-pub async fn add_source_with_invalid_url() {
+pub async fn when_add_source_with_invalid_url_expect_error() {
```
Comment (Somfic) [inline src/tests/unit/sources.rs:60]: Shouldn't it be expecting a ResourecError::InvalidInput error here? `when_add_source_with_invalid_url_expect_invalid_input`?

**Review by Somfic — COMMENTED** [review]
For tests that are checking for an error or failure, it might be helpful to specify *what* type of error or failure is expected (e.g. `when_get_not_existing_user_ok_expect_not_found
`).

**Comment by KaanMol** [comment]
> For tests that are checking for an error or failure, it might be helpful to specify *what* type of error or failure is expected (e.g. `when_get_not_existing_user_ok_expect_not_found
> 
> `).

That's the the thing, most of them do not check the actual error. I have done it in a few of mine, but the others seem to check if is_err is true. If we want to have an actual error, we'll need to check the actual error instead of is_err

**Comment by Somfic** [comment]
Reason for this was that a rewrite of the error system is pending. Should we see pass this for now or update the tests?

**Comment by KaanMol** [comment]
In the case we pass it for now, we have to create an issue, so that we do not forget it. For me it depends on when the error system will be rewritten. If that is for example within a week, lets pass it, if it is in 4 months, we might wanna update it for now. When do you think we will queue this?

**Comment by Somfic** [comment]
My take is, the sooner the less refactoring we have to do. So the priority for this is high for me. Refactoring the errors should be doable within this week.

**Comment by KaanMol** [comment]
In that case, pass it for now, and the names will be updated when the errors are implemented. Since 'expecting' error is at the moment more correct than 'expect invalid url error' for example

**Comment by Somfic** [comment]
Alrighty, need your updated review before merging.

**Review by KaanMol — APPROVED** [review]
Good to go!

**Pull request merged**